01-messaging.md

```diff
@@ -473,6 +480,53 @@ every message maximally).
 Finally, the usage of periodic `ping` messages serves to promote frequent key
 rotations as specified within [BOLT #8](08-transport.md).
 
+## Node Backup Storage
+
+Nodes that advertise the `provide_peer_backup_storage` feature offer storing
+arbitrary data for their peers. The data stored must not exceed 32000 bytes,
+which lets it fit in existing lightning messages (e.g. an `init` message).
+
+Nodes that advertise the `want_peer_backup_storage` feature want to have their
+data stored by their peers that support `provide_peer_backup_storage`.
+
+Nodes can verify that their `provide_peer_backup_storage` peers correctly store
+their backup data at each reconnection, by comparing the contents of the
+retrieved backups with the last one they sent. However, nodes should not expect
+their peers to always have their latest backup available (note that there is no
+acknowledgement to ensure a backup has been correctly stored).
+
+There are two types of backups:
+  - `node_backup`, described in this section
+  - `channel_backup`, described in [Bolt #2](02-peer-protocol.md#channel-backup-storage)
+
+Nodes ask their peers to store data using the `update_node_backup` message:
+
+1. type: 20 (`update_node_backup`)
+2. data:
+   * [`u16`: `length`]
+   * [`length*byte`:`node_backup`]
+
+A node with `want_peer_backup_storage` activated:
+  - if its peer doesn't support `provide_peer_backup_storage`:
+    - MUST NOT send `update_node_backup`
+  - otherwise:
+    - MAY send `update_node_backup` whenever necessary
+    - MUST limit its `node_backup` to 32000 bytes
+    - when it receives `init` with an outdated or missing `node_backup`:
+      - MAY send a warning
+
+A node with `provide_peer_backup_storage` activated:
+  - when it receives `update_node_backup`:
+    - if the `node_backup` exceeds 32000 bytes:
+      - SHOULD send a warning
+      - MUST NOT store this backup data
+    - otherwise:
+      - SHOULD store this backup data
```

Comment:
what if the sender does not have a channel with us? maybe discard the blob and send a warning?

Reply:
It should be up to the implementation to decide, probably need a `MAY ignore backup data from peers with whom they have no channels` or something similar

Reply:
Sounds good.

Reply:
Added and rebased in https://github.com/lightning/bolts/pull/881/commits/daa853ce357203ddebb0bc988ac54b9af60fb140